From a5594ef84ce2746fc1a115e6899350d616c6aaf8 Mon Sep 17 00:00:00 2001
From: Vineet Nayak <vnayak@nvidia.com>
Date: Wed, 17 Oct 2012 17:38:38 -0700
Subject: [PATCH 6/6] stagefright: fix crash when mFileSource is NULL

Adding check for NULL on mFileSource in onVideoEvent
for the cases where reset() is called (thus
clearing mFileSource) before onVideoEvent(). Cases
where dataSource itself is NULL should be caught in
setDataSource().

Bug 1156484

Change-Id: I0211ee8e0a51753966d033413434e3b88fd28066
---
 media/libstagefright/AwesomePlayer.cpp |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index 977b377..5240fec 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -1935,8 +1935,9 @@ void AwesomePlayer::onVideoEvent() {
         if (mSeeking != NO_SEEK) {
             ALOGV("seeking to %lld us (%.2f secs)", mSeekTimeUs, mSeekTimeUs / 1E6);
 
-            if(mDataSourceFlags & DataSource::kIsLocalDataSource)
+            if(mFileSource != NULL && (mDataSourceFlags & DataSource::kIsLocalDataSource)) {
                     mFileSource->setBypassMode(false);
+            }
             options.setSeekTo(
                     mSeekTimeUs,
                     mSeeking == SEEK_VIDEO_ONLY
-- 
1.7.1

